Portal/src/Datahub.Portal/Pages/Workspace/Settings/WorkspaceSettingsPage.razor

```diff
@@ -51,16 +86,46 @@
             .FirstOrDefaultAsync(p => p.Project_Acronym_CD == WorkspaceAcronym);
 
         _budget = project.Project_Budget ?? 0;
+
+        var currentUser = await _userInformationService.GetCurrentPortalUserAsync();
+        var _projectUsers = await _projectUserManagementService.GetProjectUsersAsync(WorkspaceAcronym);
+
+        foreach (var user in _projectUsers)
+        {
+            if (user.Role.Id.Equals(2) && user.PortalUser.Id.Equals(currentUser.Id))
+            {
+                _isDisabled = false;
+            }
```

Comment:
That's the right idea, but I'll show you the `NotAuthorized` section in the Auth view that'll simplify things